google/cloud/speech/client.py

```diff
@@ -0,0 +1,168 @@
+        :param sample_rate: Sample rate in Hertz of the audio data sent in all
+                           RecognitionAudio messages. Valid values are: 8000-
+                           48000. 16000 is optimal. For best results, set the
+                           sampling rate of the audio source to 16000 Hz.
+                           If that's not possible, use the native sample rate
+                           of the audio source (instead of re-sampling).
+
+        :type language_code: str
+        :param language_code: (Optional) The language of the supplied audio as
+                             BCP-47 language tag. Example: "en-GB".
+                             If omitted, defaults to "en-US".
+
+        :type max_alternatives: int
+        :param max_alternatives: (Optional) Maximum number of recognition
+                                hypotheses to be returned. The server may
+                                return fewer than maxAlternatives.
+                                Valid values are 0-30. A value of 0 or 1
+                                will return a maximum of 1. Defaults to 1
+
+        :type profanity_filter: bool
+        :param profanity_filter: If True, the server will attempt to filter
+                                out profanities, replacing all but the
+                                initialcharacter in each filtered word with
+                                asterisks, e.g. "f***". If False or omitted,
+                                profanities won't be filtered out.
+
+        :type speech_context: list
+        :param speech_context: A list of strings (max 50) containing words and
+                              phrases "hints" so that the speech recognition
+                              is more likely to recognize them. This can be
+                              used to improve the accuracy for specific words
+                              and phrases. This can also be used to add new
+                              words to the vocabulary of the recognizer.
+
+        :rtype: list
+        :returns: A list of tuples. One tuple for each alternative. Each tuple
+                  contains a transcript text and a confidence value (between
+                  0.0 and 1.0)
+        """
+
+        if (content is None) and (uri is None):
+            message = 'content and uri cannot be both equal to None'
+            raise ValueError(message)
+
+        if (content is not None) and (uri is not None):
+            message = 'content and uri cannot be both different from None'
+            raise ValueError(message)
+
+        if content is not None:
+            audio = {'content': content}
+        else:
+            audio = {'uri': uri}
+
+        required_params = [('encoding', encoding), ('sampleRate', sample_rate)]
+        for param_name, param in required_params:
+            if param is None:
+                message = '%r cannot be None' % (param_name)
+                raise ValueError(message)
```

Comment:
Rather than looping, just check these with individual `if` statements:  move them up below the other validation tests above.